src/util/TchapClientUtils.ts

```diff
@@ -0,0 +1,99 @@
+import BasePlatform from "matrix-react-sdk/src/BasePlatform";
+import { MatrixClientPeg } from "matrix-react-sdk/src/MatrixClientPeg";
+import PlatformPeg from "matrix-react-sdk/src/PlatformPeg";
+
+export default class TchapClientUtils {
+    static SYNC_STORE_NAME = "matrix-js-sdk:riot-web-sync";
+
+    /**
+     * Determine weither the app needs a refresh after loading
+     * @param indexedDB the indexDb interface
+     * @returns Promise(true) if a refresh is needed, Promise(false) in other cases
+     */
+    public static async doesNeedRefresh(indexedDB: IDBFactory): Promise<boolean> {
+        if (!indexedDB) {
+            Promise.reject("indexDb is undefined");
+        }
+        //read version of tchap app in localstorage, if it does start with 4, don't refresh
+        if (TchapClientUtils.getAppVersion() && TchapClientUtils.getAppVersion().startsWith("4")) {
+            return Promise.resolve(false);
+        }
+
+        try {
+            //read version of matrix-js-sdk:riot-web-sync, if it is older than version 4, refresh is needed
+            const version = await TchapClientUtils.getStoreVersion(indexedDB, TchapClientUtils.SYNC_STORE_NAME);
+            return Promise.resolve(version < 4);
+        } catch (error) {
+            //if store did not exists or a technical error occured do not refresh (safety first)
+            console.warn(error);
+            return Promise.resolve(false);
+        }
+    }
+
+    /**
+     * replicate the behaviour of the button Clear Cache and Reload
+    * https://github.com/matrix-org/matrix-react-sdk/blob/3c5c2bef6dbac51ce6e1864056523815ca4c38d9/src/components/views/settings/tabs/user/HelpUserSettingsTab.tsx#L308
+     * @returns nothing
+     */
+    public static clearCacheAndReload(): void {
+        console.log(":TCHAP: clearCacheAndReload at client startup");
+
+        if (!PlatformPeg.get()) return;
+
+        MatrixClientPeg.get().stopClient();
+        MatrixClientPeg.get().store.deleteAllData().then(() => {
```

Comment:
what data does this delete ? 
Is this a problem for users with a single session, who don't have other sessions to get keys from ? (we don't have secure storage in tchap)

(just asking general questions, to make sure we're not doing something crazy)